src/scripts/breeding/Breeding.ts

```diff
@@ -285,7 +285,8 @@ class Breeding implements Feature {
             this._eggList[index](new Egg());
             this.moveEggs();
             if (this._queueList().length) {
-                const nextEgg = this.createEgg(this._queueList.shift());
+                const pokemon = PokemonHelper.getPokemonById(this._queueList.shift());
+                const nextEgg = this.createEgg(pokemon.name);
```

Comment:
I think createEgg should take the id as a parameter. Yes, you need to change createFossilEgg and createTypedEgg to fetch the pokemon first, but this code happens way more often, so removing a lookup here, should be better